### PR TITLE
RDS Events support

### PR DESCRIPTION
**Pull request by limod**
#### Summary
I added the processing of [RDS Events](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_Events.html) since they work over SNS subscription too and i found no alternative plugin.
This is my first experience with go, feedback welcome. Also i think this project has no tests i should adapt?

### CODE REVIEW

File: server/sns.go [hunk 1]
```diff
@@ -61,3 +61,12 @@ type SNSMessageNotification struct {
 		EvaluateLowSampleCountPercentile string  `json:"EvaluateLowSampleCountPercentile"`
 	} `json:"Trigger"`
 }
+
+type SNSRdsEventNotification struct {
+	EventSource string `json:"Event Source"`
+	EventTime string `json:"Event Time"`
```
Comment (hanzei) [inline server/sns.go:67]: You need to `gofmt` these lines

**Review by mickmister — APPROVED** [review]
LGTM :+1: Thanks @limod!

**Comment by mickmister** [comment]
> Also I think this project has no tests i should adapt?

@limod Feel free to add tests as you see fit. Any code that is entirely webhook driven should be pretty testable.

File: server/plugin.go [hunk 2]
```diff
@@ -153,16 +153,74 @@ func (p *Plugin) handleNotification(body io.Reader) {
 		return
 	}
 
+	var attachment model.SlackAttachment
```
Comment (jfrerich) [inline server/plugin.go:156]: I'm wondering if a small refactor would benefit this code.  Functionally, I believe you are correct, but think something like the following would help clarify and allow for extending to other notification types.  I think removing this `if/else` block style would be beneficial for readability.

I would create a smaller `sendPostNotification` function containing `post` initialization and the `model.ParseSlackAttachement()` and `CreatePost` calls. 

Something like the following: 
```go
	isRdsEvent, messageNotification := p.isRDSEvent(notification.Message)
	if isRdsEvent {
		p.API.LogDebug("Processing RDS Event")
		p.sendPostNotification(p.createSNSRdsEventAttachment(notification.Subject, messageNotification))
		return
	}

	if isAlarm, messageNotification := p.isCloudWatchAlarm(notification.Message); isAlarm {
		p.API.LogDebug("Processing CloudWatch alarm")
		p.sendPostNotification(p.createSNSMessageNotificationAttachment(notification.Subject, messageNotification))
		return 
	}


func (p *Plugin) sendPostNotification(attachment model.SlackAttachment) {
	post := &model.Post{
		ChannelId: p.ChannelID,
		UserId:    p.BotUserID,
	}
	model.ParseSlackAttachment(post, []*model.SlackAttachment{&attachment})
	if _, appErr := p.API.CreatePost(post); appErr != nil {
		return
	}
}



```

**Review by jfrerich — CHANGES_REQUESTED** [review]
Thanks for the PR, @limod!  I have one change request.  What do you think about my suggestion?  cc: @hanzei

File: server/plugin.go (hunk 2 above)
Comment (limod) [reply server/plugin.go]: Thanks @jfrerich , suggestion is implemented!

**Review by jfrerich — APPROVED** [review]
Awesome addition to the plugin, @limod!  Also, thanks for the super quick refactor!

**Comment by jfrerich** [comment]
@DHaussermann, gentle ping for review

**Comment by limod** [comment]
Is this PR dead?

**Comment by jasonblais** [comment]
@DHaussermann Kind reminder with a review here - or delegate to another QA if you currently don't have the time?

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- RDS events delivered via subscription to an existing topic will be posted in the channel as expected 
- Various RDS events are delivering successfully 
- Posts made by the bot show clear `RDS Notification Message` harder and the data and links are structured clearly. 

LGTM!

Very sorry @limod for the long delay is testing this. Thank you for this enhancement. Very much appreciated!

**Pull request merged**